lib/improver/tests/wind_downscaling/test_roughnesscorrection.py

```diff
@@ -729,6 +737,17 @@ def test_section2c(self):
             _ = multip_hc_rc.run_hc_rc([uin, uin], dtime=2, height=heights,
                                        aslist=True)
 
+    def test_section2d(self):
+        """ Test whether output is float32. """
+        hlvs = 10
+        uin = np.ones(hlvs)*20
+        heights = ((np.arange(hlvs)+1)**2.)*12.
+        multip_hc_rc = TestMultiPoint(3)
+        land_hc_rc = multip_hc_rc.run_hc_rc(uin, dtime=1, height=heights)
+        hidx = land_hc_rc.shape.index(hlvs)
+        for field in land_hc_rc.slices_over(hidx):
```

Comment:
Is there a reason to slice over `hidx` in this case? I would have thought that:
```python
        hidx = land_hc_rc.shape.index(hlvs)
        for field in land_hc_rc.slices_over(hidx):
            self.assertEqual(field.dtype, np.float32)
```
could be replaced by:
```python
self.assertEqual(land_hc_rc.dtype, np.float32)
```